nf_core/modules/lint/module_changes.py

```diff
@@ -39,10 +40,12 @@ def module_changes(module_lint_object, module):
             return
     else:
         tempdir = module.module_dir
+    module.branch = module_lint_object.modules_json.get_component_branch(
+        "modules", module.module_name, module.repo_url, module.org
+    )
+    modules_repo = nf_core.modules.modules_repo.ModulesRepo(remote_url=module.repo_url, branch=module.branch)
 
-    for f, same in module_lint_object.modules_repo.module_files_identical(
-        module.module_name, tempdir, module.git_sha
-    ).items():
+    for f, same in modules_repo.module_files_identical(module.module_dir, tempdir, module.git_sha).items():
```

Comment:
This `module.module_dir` is a temporary directory, so not the same as we were obtaining before inside this function with `self.get_component_dir(module_name, "modules")`
Going back to passing the module name and obtaining the directory inside the function should make the test pass

Reply:
For the record, if we read this in the future, `module.module_dir` is the module directory in the pipeline, while what we need is the module directory in the modules repo.